src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts

```diff
@@ -28,7 +27,7 @@ export class RunInsightQueryAction extends Action {
 			if (typeof (context.insight.query) === 'string') {
 				queryString = context.insight.query;
 			} else {
-				queryString = context.insight.query.join(os.EOL);
+				queryString = context.insight.query.join('\n');
```

Comment:
You should get the newline from the TextResourcePropertiesService https://github.com/microsoft/azuredatastudio/blob/eng/linux/src/vs/workbench/services/textresourceProperties/common/textResourcePropertiesService.ts#L20